src/cryptsetup/cryptsetup.c

```diff
@@ -710,7 +710,8 @@ static int run(int argc, char *argv[]) {
                                                          flags);
                         if (r >= 0)
                                 break;
-                        if (r != -EAGAIN)
+                        /* Avoid immediately failing due to key file issues */
+                        if (r != -EAGAIN && !key_file)
```

Comment:
Hmm, I think it would be better to change attach_luks_or_plain() instead. It has the `crypt_activate_by_keyfile_offset()` in the middle, that is used when a key_file path is supplied. When it returns we turn EPERM into EAGAIN right now with a friendly log msg. I think it we should add another if check right there, that turns EINVAL into EAGAIN too with a different, appropriate error msg. EINVAL is after all the (awfully generic) error libcryptsetup returns if the keyfile doesn't exist. 

Or in other words: I think we should handle this error as early as possible, with a friendly log msgs, instead of doing so after it already propagated further through our stack losing its context.

The error msg should probably suggest what the context of all of is this i.e. `Failed to activate with key file '%s'. (Key file missing?)` or so. (since EINVAL is so generic we can only guess, hence the question in the end.